Libraries/src/Amazon.Lambda.Annotations.SourceGenerator/FileIO/DirectoryManager.cs

```diff
@@ -16,29 +16,34 @@ public class DirectoryManager : IDirectoryManager
         /// </summary>
         public string GetRelativePath(string relativeTo, string path)
         {
-            relativeTo = SanitizePath(relativeTo);
-            path = SanitizePath(path);
+            if (string.IsNullOrEmpty(relativeTo) || string.IsNullOrEmpty(path))
+            {
+                throw new InvalidOperationException($"{nameof(relativeTo)} or {nameof(path)} cannot be null or empty");
+            }
+
+            // If there is no common ancestor, just return the path.
+            if (!string.Equals(SanitizePath(relativeTo).Split('/')[0], SanitizePath(path).Split('/')[0]))
+            {
+                return path;
+            }
+
+            // Calculate the full paths by preserving the OS specific directory separator character and then perform the sanitization.
+            relativeTo = SanitizePath(new DirectoryInfo(relativeTo).FullName);
```

Comment:
If you move these 2 lines above the if block on line 25 you can remove the `SanitizePath` calls in the if condition.

Reply:
Per offline conversations I understand this is intentional because we need to do the comparison before calling `new DirectoryInfo().FullName`.